### DISCUSSION TITLE
CI binaries ECONNREFUSED

### DISCUSSION THREAD

**Discussion by yaniv-peretz** [prisma/prisma#12195] [Q&A] [answered]
I'm installing prisma as part of ci (Jenkins, using Prisma to created test data).
```
> npm --ignore-scripts i
> npx prisma db pull
Error: request to https://binaries.prisma.sh/all_commits/73e60b76d394f8d37d8ebd1f8918c79029f0db86/debian-openssl-1.1.x/migration-engine.gz.sha256 failed, reason: connect ECONNREFUSED 108.156.107.28:443
```

Is this a request for a checksum value?
Can I skip / ignore this step?
Or what else can I do to satisfy Jenkins?

At the moment due to this issue, I'm restricted from using Prisma (since our CI can't support it yet).

**Comment by pantharshit00** [comment, marked as answer]
You can upload custom binaries in this case. Read more about that here: https://www.prisma.io/docs/concepts/components/prisma-engines#using-custom-engine-libraries-or-binaries

**Reply by yaniv-peretz** [reply]
I tried adding these binaries into Git (330Mb), in this case I didn't need to download the binaries, and the checksum.
It worked, but I was requested to remove them due to the size (we are only using Prisma for tests),

I guess maybe https://www.prisma.io/docs/concepts/components/prisma-engines#hosting-engines
could work, but then I'll need to host them on our repo.

Thing is, why can't I just skip the chcksum (sha)?

**Reply by pantharshit00** [reply]
I will suggest you to open a new feature request for skipping SHA checks here: https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=

No one has asked for this yet so it wasn't added, but we can obviously add it if it makes sense.

**Reply by yaniv-peretz** [reply]
https://github.com/prisma/prisma/issues/12230

**Comment by janpio** [comment]
Taking a step back here, why are you forbidden to request this file @Yanipo5? Just no external downloads at all in your environment?

**Reply by yaniv-peretz** [reply]
It's a CI environment, of which I don't have full control over. It's probably an enterprise policy.
In other words, locally everything works fine,
But then in our pre-merge CI (Jenkins), installing Prisma fails due to checksum as external urls are forbiden in CI.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
